### DISCUSSION TITLE
How to throw a custom Error in Access Control Options

### DISCUSSION THREAD

**Discussion by ogheneovo12** [keystonejs/keystone#8929] [Questions]
i have a `isSignedIn` access control methods, basically helps to check if a session actually exists for any operations, normally all Access Control Will return the KS_ACCESS_DENIED error, which in my case is quite generic, so i need a way to create a custom Error code i.e KS_UNAUTHORIZED, so this way on my seperate client, i can always log the user out, if and only no session exist (not when there's an actuall RBAC based error).

throwing a normal error just uses the KS_EXTENSION_ERROR field.

isSignedInMethod
```
export const isSignedIn =
  (throwError = true) =>
  ({ session }: AccessArgs) => {
    const isLoggedIn = Boolean(session);
    if (!isLoggedIn && throwError) {
      throw Error("UNAUTHORIZED");
    }
    return isLoggedIn;
  };

```

Usage 
```
const Magazine: Lists.Magazine = list({
  access: {
    operation: {
      ...allOperations(permissions.canManageMagazines),
      query: isSignedIn(),
    },
  },
  
 ...
 })
 
```

Error Response

![Screenshot from 2023-11-22 18-51-42](https://github.com/keystonejs/keystone/assets/49083654/eff28b52-a619-48d0-982c-7a9792d67577)

**Comment by kennedybaird** [comment]
Did you make progress on this? I was just playing in this area today, didn't get a custom error showing unfortunately.
